### PR TITLE
chore: added noImplicitOverride option to config

### PR DESCRIPTION
**Pull request by YaTut1901**
- Closes https://github.com/FuelLabs/fuels-ts/issues/3275

# Checklist

- [x] All **changes** are **covered** by **tests** (or not applicable)
- [x] All **changes** are **documented** (or not applicable)
- [x] I **reviewed** the **entire PR** myself (preferably, on GH UI)
- [x] I **described** all **Breaking Changes** (or there's none)

### CODE REVIEW

File: .changeset/strange-starfishes-raise.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+---
+"@fuel-ts/account": patch
+"@fuel-ts/address": patch
+"@fuel-ts/program": patch
+"@fuel-ts/script": patch
+"fuels": patch
+---
+
+chore(packages): added noImplicitOverride option to config
```
Comment (petertonysmith94) [inline .changeset/strange-starfishes-raise.md:9]:
```suggestion
chore: added noImplicitOverride option to config
```

File: packages/abi-coder/src/encoding/coders/EnumCoder.ts [hunk 2]
```diff
@@ -21,8 +21,8 @@ export class EnumCoder<TCoders extends Record<string, Coder>> extends Coder<
   InputValueOf<TCoders>,
   DecodedValueOf<TCoders>
 > {
-  name: string;
-  coders: TCoders;
+  public override name: string;
+  public coders: TCoders;
```
Comment (petertonysmith94) [inline packages/abi-coder/src/encoding/coders/EnumCoder.ts:25]:
```suggestion
  override name: string;
  coders: TCoders;
```

Why change the access visibility here?

File: packages/abi-coder/src/encoding/coders/StructCoder.ts [hunk 3]
```diff
@@ -18,8 +18,8 @@ export class StructCoder<TCoders extends Record<string, Coder>> extends Coder<
   InputValueOf<TCoders>,
   DecodedValueOf<TCoders>
 > {
-  name: string;
-  coders: TCoders;
+  public override name: string;
+  public coders: TCoders;
```
Comment (petertonysmith94) [inline packages/abi-coder/src/encoding/coders/StructCoder.ts:22]:
```suggestion
  override name: string;
  coders: TCoders;
```

File: packages/abi-coder/src/encoding/coders/OptionCoder.ts [hunk 4]
```diff
@@ -6,7 +6,7 @@ type SwayOption<T> = { None: [] } | { Some: T };
 export type Option<T> = T | undefined;
 
 export class OptionCoder<TCoders extends Record<string, Coder>> extends EnumCoder<TCoders> {
-  encode(value?: Option<unknown>): Uint8Array {
+  public override encode(value?: Option<unknown>): Uint8Array {
```
Comment (petertonysmith94) [inline packages/abi-coder/src/encoding/coders/OptionCoder.ts:9]:
```suggestion
  override encode(value?: Option<unknown>): Uint8Array {
```

File: packages/abi-coder/src/encoding/coders/OptionCoder.ts [hunk 5]
```diff
@@ -19,7 +19,7 @@ export class OptionCoder<TCoders extends Record<string, Coder>> extends EnumCode
     return { None: [] };
   }
 
-  decode(data: Uint8Array, offset: number): [DecodedValueOf<TCoders>, number] {
+  public override decode(data: Uint8Array, offset: number): [DecodedValueOf<TCoders>, number] {
```
Comment (petertonysmith94) [inline packages/abi-coder/src/encoding/coders/OptionCoder.ts:22]:
```suggestion
  override decode(data: Uint8Array, offset: number): [DecodedValueOf<TCoders>, number] {
```

**Review by petertonysmith94 — APPROVED** [review]

File: .changeset/strange-starfishes-raise.md [hunk 6]
```diff
@@ -0,0 +1,9 @@
+---
+"@fuel-ts/account": patch
+"@fuel-ts/address": patch
+"@fuel-ts/program": patch
+"@fuel-ts/script": patch
```
Comment (danielbate) [inline .changeset/strange-starfishes-raise.md:5]: Please regenerate the changeset as this looks like it's missing packages.

**Review by danielbate — CHANGES_REQUESTED** [review]

File: .changeset/strange-starfishes-raise.md [hunk 7]
```diff
@@ -0,0 +1,9 @@
...
+"fuels": patch
+---
+
+chore: added noImplicitOverride option to config
```
Comment (Torres-ssf) [inline .changeset/strange-starfishes-raise.md:9]: @YaTut1901 It appears that we have 2 changeset files being added. We can safely remove `strange-starfishes-raise.md`. 

`large-points-arrive.md` seems to be already correct

**Review by Torres-ssf — CHANGES_REQUESTED** [review]

File: .changeset/strange-starfishes-raise.md
Comment (petertonysmith94) [inline .changeset/strange-starfishes-raise.md:1]: @YaTut1901 please delete this changeset :)

**Review by petertonysmith94 — APPROVED** [review]

**Review by Torres-ssf — APPROVED** [review]

**Review by maschad — APPROVED** [review]
Thanks for the great work @YaTut1901 🚀

**Pull request merged**